src/cljc/rems/common/application_util.cljc

```diff
@@ -69,27 +70,37 @@
          (sort (mapv parse-sortable-external-id ["ABC/2000.1" "ABC/2002.0" "ABC/2002.2"  "ABC/2302.0" "ABC/2000.2" "ABC/2000.3"])))))
 
 (defn can-redact-attachment [attachment roles userid]
-  (when-not (:attachment/redacted attachment)
+  (let [event-id (get-in attachment [:attachment/event :event/id])
+        allowed-roles (:attachment/redact-roles attachment)]
     (cond
-      (nil? (get-in attachment [:attachment/event :event/id]))
+      (or (:attachment/redacted attachment)
+          (not event-id))
       false
 
-      (some roles (:attachment/redact-roles attachment))
-      true
+      (not= userid (get-in attachment [:attachment/user :userid]))
+      (some? (not-empty (clojure.set/intersection (set roles)
```

Comment:
As it seems explicit here that we aim to return a boolean always, we could name this function like normal predicates `can-redact-attachment?`. Otherwise we could just use `seq` here.

Reply:
I think there's also some other possibilities for golfing for simplicity in the whole `cond`.

Reply:
An idea
```clj
(defn can-redact-attachment? [attachment roles userid]
  (let [event-id (get-in attachment [:attachment/event :event/id])
        allowed-roles (:attachment/redact-roles attachment)]
    (cond
      (:attachment/redacted attachment) false

      (not event-id) false

      (= userid (get-in attachment [:attachment/user :userid]))
      true

      (empty? (clojure.set/intersection (set roles)
                                        (set allowed-roles)))
      false

      :else true)))```

Reply:
yeah, that could work!

i like `cond` because it makes multiple cases more readable, but sometimes i'm not sure which way reads better, explicit return values for each case or function call returning value. like here, `(not (empty? ...))` is discouraged by clj-kondo (warning) so i tend to not use it, but it is more readable than `(some? (not-empty ...))` when only boolean is required.

Reply:
Yeah, I like `cond` too, just have to organize it so that the clauses are simple and make sense.